### ISSUE TITLE
[Feature request] Configurable invincibility ticks upon rolling

### ISSUE THREAD

**Issue by Warhand** [ZsoltMolnarrr/Rolling#9]
**I read project readme, checked client side and server side configurations, and no such feature exists** - YES/NO
Yes
**Is your feature request related to a problem? Please describe.**
N/a
**Describe the solution you'd like**
The introduction of I-frames as a config option would make this mod perfect for what I have in mind, which is basically me looking for a replica of the dark souls dodgeroll.
TL:DR; I think it would be neat if the player was invincible for a period of time while rolling, to make rolling more useful.

**Describe alternatives you've considered**
I'm not really sure if there are any other alternatives to achieve the same feel in combat.

**Comment by ZsoltMolnarrr** [comment]
Hello!
There are 2 issues with this suggestion:

1. I am not against this feature, however I am not interested implementing it either.
(This is an open source project, anyone is free to implement it, given the submitted code meets the quality standards.)

2. Better Combat bypasses the vanilla implementation of invincibility frames, to allow fast weapons working. Without it, dual wielding would have not too much point.
To make this feature work, either Better Combat needs to be touched, or a completely separate i-frame implementation has to be added by this mod.

**Comment by Warhand** [comment]
I see... thats unfortunate but fair. I don't really have any coding experience, nor do I have the time for it, so hopefully someone else happens upon the feature suggestion and implements it themselves.

**Comment by onspeed8** [comment]
Lacking the technical knowhow to do this myself i figured i'd leave this tiny nugget of information wich may help someone implement this feature. (all be it slightly "hacky")

- Resistance V gives the player full immunity to most damage types (no explosion/starvation/void)
- Potion effects are able to last less then a second (at least when using command blocks, i'm assuming mods are even more flexible) [Example](https://www.minecraftforum.net/forums/minecraft-java-edition/redstone-discussion-and/2586207-potion-effects-last-less-than-1-second)

So the way this could be implemented is by giving the player a short burst of resistance V during the roll

_________
Potential issue: When the player already has resistance it'll probably override it and take it away after the roll
Potential solution: Check current player's potion effects and save **Resistance** level + duration and reappy them after the roll
_________
Downside: Resistance will be visible (particles / HUD)

**Comment by Darkmega18** [comment]
what use is a dodge roll without i-frames to avoid cheese hits in a game where attacks are usually instant? :V

**Comment by ZsoltMolnarrr** [comment]
Will be available in 1.3.0.

**Comment by Darkmega18** [comment]
oh, awesome. Thanks. :D
